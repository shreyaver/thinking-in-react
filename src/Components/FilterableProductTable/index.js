import React, { Component } from 'react';
import SearchBar from '../SearchBar';
import ProductTable from '../ProductTable';
import PropTypes from 'prop-types';

class FilterableProductTable extends Component {
    state = { filterText: this.props.filterText, inStockOnly: this.props.inStockOnly }
    changeFilterTextState = (newFilterText) => {
        this.setState((state) => ({
            filterText: newFilterText
        }))
    }
    changeInStockOnlyState = (newInStockOnly) => {
        this.setState((state) => ({
            inStockOnly: newInStockOnly
        }))
    }
    render() {
      return (
          <div>
              <SearchBar filterText = {this.state.filterText} inStockOnly = {this.state.inStockOnly} changeFilterTextState = {this.changeFilterTextState} changeInStockOnlyState = {this.changeInStockOnlyState}/>
              <ProductTable productArray = {this.props.productArray} filterText = {this.state.filterText} inStockOnly = {this.state.inStockOnly} />
          </div>
      ) 
    }
}
FilterableProductTable.propTypes = {
    filterText: PropTypes.string,
    inStockOnly: PropTypes.bool
  };
FilterableProductTable.defaultProps = {
    filterText: '',
    inStockOnly: false
};
export default FilterableProductTable;