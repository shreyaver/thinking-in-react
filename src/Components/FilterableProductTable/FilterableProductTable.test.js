import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import FilterableProductTable from '.';

describe('FilterableProductTable', () => {
  const productArray = [
    {
      category: 'Sporting Goods', price: '$49.99', stocked: true, name: 'Football',
    },
    {
      category: 'Sporting Goods', price: '$9.99', stocked: true, name: 'Baseball',
    },
    {
      category: 'Sporting Goods', price: '$29.99', stocked: false, name: 'Basketball',
    },
    {
      category: 'Electronics', price: '$99.99', stocked: true, name: 'iPod Touch',
    },
    {
      category: 'Electronics', price: '$399.99', stocked: false, name: 'iPhone 5',
    },
    {
      category: 'Electronics', price: '$199.99', stocked: true, name: 'Nexus 7',
    },
  ];
  it('renders without crashing for stocked and unstocked products and no filter text', () => {
    const filterableProductTable = renderer.create(<FilterableProductTable productArray={productArray} />).toJSON();
    expect(filterableProductTable).toMatchSnapshot();
  });
  it('gets state change when checkbox is toggled', () => {
    const wrapper = mount(<FilterableProductTable productArray={productArray} />);
    expect(wrapper.instance().state.inStockOnly).toEqual(false);
    wrapper.find('.InStockCheckBox').simulate('change', { target: { checked: true } });
    expect(wrapper.instance().state.inStockOnly).toEqual(true);
    wrapper.unmount();
  });
  it('gets state change when text is entered in searchbox', () => {
    const wrapper = mount(<FilterableProductTable productArray={productArray} />);
    expect(wrapper.instance().state.filterText).toEqual('');
    wrapper.find('.FilterSearchBox').simulate('change', { target: { value: 'abc' } });
    expect(wrapper.instance().state.filterText).toEqual('abc');
    wrapper.unmount();
  });
});
// xit('renders without crashing for only stocked products and no filter text', () => {
//   const filterableProductTable = renderer.create(<FilterableProductTable productArray = {productArray} filterText= '' inStockOnly= {true} />).toJSON();
//   expect(filterableProductTable).toMatchSnapshot();
// });
// xit('renders without crashing for stocked and unstocked products and passed filter text', () => {
//   const filterableProductTable = renderer.create(<FilterableProductTable productArray = {productArray} filterText= 'Phone' inStockOnly= {false} />).toJSON();
//   expect(filterableProductTable).toMatchSnapshot();
// });
// xit('renders without crashing for only stocked products and passed filter text', () => {
//   const filterableProductTable = renderer.create(<FilterableProductTable productArray = {productArray} filterText= 'ball' inStockOnly= {true} />).toJSON();
//   expect(filterableProductTable).toMatchSnapshot();
// });
