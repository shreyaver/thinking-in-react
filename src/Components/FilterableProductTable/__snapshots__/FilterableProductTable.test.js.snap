// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterableProductTable renders without crashing for stocked and unstocked products and no filter text 1`] = `
<div>
  <form>
    <input
      className="FilterSearchBox"
      onChange={[Function]}
      placeholder="Search..."
      type="text"
      value=""
    />
    <br />
    <input
      checked={false}
      className="InStockCheckBox"
      onChange={[Function]}
      type="checkbox"
    />
    Only show products in stock
  </form>
  <table
    style={
      Object {
        "border": "1px solid black",
      }
    }
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Price
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          colSpan="2"
        >
          Sporting Goods
        </th>
      </tr>
      <tr>
        <td
          className="ProductName"
          style={
            Object {
              "color": "black",
            }
          }
        >
          Football
        </td>
        <td>
          $49.99
        </td>
      </tr>
      <tr>
        <td
          className="ProductName"
          style={
            Object {
              "color": "black",
            }
          }
        >
          Baseball
        </td>
        <td>
          $9.99
        </td>
      </tr>
      <tr>
        <td
          className="ProductName"
          style={
            Object {
              "color": "red",
            }
          }
        >
          Basketball
        </td>
        <td>
          $29.99
        </td>
      </tr>
      <tr>
        <th
          colSpan="2"
        >
          Electronics
        </th>
      </tr>
      <tr>
        <td
          className="ProductName"
          style={
            Object {
              "color": "black",
            }
          }
        >
          iPod Touch
        </td>
        <td>
          $99.99
        </td>
      </tr>
      <tr>
        <td
          className="ProductName"
          style={
            Object {
              "color": "red",
            }
          }
        >
          iPhone 5
        </td>
        <td>
          $399.99
        </td>
      </tr>
      <tr>
        <td
          className="ProductName"
          style={
            Object {
              "color": "black",
            }
          }
        >
          Nexus 7
        </td>
        <td>
          $199.99
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
